### PR TITLE
rtl support was added

### PR DESCRIPTION
**Pull request by shamloo**

### CODE REVIEW

**Comment by caprica** [comment]
Is there a live demo somewhere?

File: demo/simple - rtl.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+            <h1>JQuery/CSS Organisation Chart</h1>
+
+            <div id="main">
+            </div>
+
+            <div class="text">
+                <h2>JQuery</h2>
+                <pre>
+$("#organisation").orgChart({container: $("#main")});
```
Comment (caprica) [inline demo/simple - rtl.html]: Your new direction setting is missing from here.

File: demo/simple.html [hunk 2]
```diff
@@ -9,7 +9,7 @@
         <script src="../jquery.orgchart.js"></script>
         <script>
         $(function() {
-            $("#organisation").orgChart({container: $("#main")});
+            $("#organisation").orgChart({container: $("#main") , direction:'ltr'});
```
Comment (caprica) [inline demo/simple.html]: I should not need to specify direction LTR here, it's supposed to be the default.

File: demo/simple.html [hunk 3]
```diff
@@ -19,7 +19,9 @@
         <div id="left">
 
             <ul id="organisation">
-                <li><adjunct>Alfred</adjunct><em>Batman</em>
+                <li>
+                <adjunct>Alfred</adjunct>
+                <em>Batman</em>
```
Comment (caprica) [inline demo/simple.html]: What is this change? It seems unrelated to your patch.

File: jquery.orgchart.css [hunk 4]
```diff
@@ -79,21 +79,27 @@ div.orgChart.interactive div.hasChildren {
 div.orgChart div.node.hover {
     background-color      : white;
 }
+/* rtl */
+div.orgChart.ltr{ direction: ltr; }
+div.orgChart.rtl{ direction: rtl; }
 
-div.orgChart div.adjunct.node {
+div.orgChart.ltr div.adjunct.node , div.orgChart.rtl div.adjunct.node {
```
Comment (caprica) [inline jquery.orgchart.css]: These .ltr and .rtl selectors seem not necessary to me here. They have the same behaviour here.

File: jquery.orgchart.css [hunk 5]
```diff
@@ -79,21 +79,27 @@ div.orgChart.interactive div.hasChildren {
...
     position              : absolute;
     margin-top            : 8px;
-    margin-left           : -110px;
```
Comment (caprica) [inline jquery.orgchart.css]: Why?

File: jquery.orgchart.css [hunk 6]
```diff
@@ -79,21 +79,27 @@ div.orgChart.interactive div.hasChildren {
...
 }
 
-div.orgChart div.adjunct-link {
+div.orgChart.ltr div.adjunct.node{	margin-left		: -110px; }
+div.orgChart.rtl div.adjunct.node{ margin-right	: -110px; }
+
+div.orgChart.ltr div.adjunct-link , div.orgChart.rtl div.adjunct-link{
     display               : inline-block;
     position              : absolute;
-    margin-left           : -20px;
```
Comment (caprica) [inline jquery.orgchart.css]: Why?

File: jquery.orgchart.css [hunk 7]
```diff
@@ -79,21 +79,27 @@ div.orgChart.interactive div.hasChildren {
...
     margin-top            : 8px;
-    margin-left           : -110px;
     width                 : 80px;
     height                : 40px;
     background-color      : #efefef;
 }
 
-div.orgChart div.adjunct-link {
+div.orgChart.ltr div.adjunct.node{	margin-left		: -110px; }
+div.orgChart.rtl div.adjunct.node{ margin-right	: -110px; }
```
Comment (caprica) [inline jquery.orgchart.css]: Code formatting does not match existing style here.

File: jquery.orgchart.css [hunk 8]
```diff
@@ -79,21 +79,27 @@ div.orgChart.interactive div.hasChildren {
...
+div.orgChart.ltr div.adjunct-link , div.orgChart.rtl div.adjunct-link{
     display               : inline-block;
     position              : absolute;
-    margin-left           : -20px;
     margin-top            : 25px;
     width                 : 20px;
     border                : 1px dashed black;
 }
+div.orgChart.ltr div.adjunct-link{ margin-left		: -20px;}
+div.orgChart.rtl div.adjunct-link{ margin-right	: -20px;}
```
Comment (caprica) [inline jquery.orgchart.css]: Code formatting does not match existing style here.

File: jquery.orgchart.js [hunk 9]
```diff
@@ -28,7 +28,14 @@
             // Store the original element
             $this.data("chart-source", $chartSource);
             // Build the chart...
-            var $container = $("<div class='" + opts.chartClass + "'/>");
+            var chart_cls = '';
+            if(opts.direction == 'rtl'){
+                chart_cls = opts.chartClass+' rtl';
+            }else{
+                chart_cls = opts.chartClass+' ltr';
+            }
```
Comment (caprica) [inline jquery.orgchart.js]: Ternary operator would be better here, in one line.
chart_cls is not a conventional variable name, chartClass would be better.

File: jquery.orgchart.js [hunk 10]
```diff
@@ -159,8 +167,13 @@
...
                 var $downLineLine = $("<tr/>").addClass("lines x");
-                var $downLeft = $("<td>").addClass("line left");
-                var $downRight = $("<td>").addClass("line right");
+                if(opts.direction == 'rtl'){
+                    var $downLeft = $("<td>").addClass("line right");
+                    var $downRight = $("<td>").addClass("line left");
+                }else{
+                    var $downLeft = $("<td>").addClass("line left");
+                    var $downRight = $("<td>").addClass("line right");
+                }
```
Comment (caprica) [inline jquery.orgchart.js]: Code formatting does not match existing style here.

File: jquery.orgchart.js [hunk 11]
```diff
@@ -183,9 +196,14 @@
...
-                    var $right = $("<td/>").addClass("line right top");
-                    $linesRow.append($left).append($right);
+                    if(opts.direction == 'rtl'){
+                        var $left = $("<td/>").addClass("line right top");
+                        var $right = $("<td/>").addClass("line left top");
+                    }else{
+                        var $left = $("<td/>").addClass("line left top");
+                        var $right = $("<td/>").addClass("line right top");
+                    }
+                      $linesRow.append($left).append($right);
```
Comment (caprica) [inline jquery.orgchart.js]: Code formatting does not match existing style here.

File: jquery.orgchart.js [hunk 12]
```diff
@@ -215,4 +233,4 @@
         $appendTo.append($table);
     };
 
-})(jQuery);
+})(jQuery);
```
Comment (caprica) [inline jquery.orgchart.js]: Don't remove the newline at the EOF.

Comment (caprica) [inline jquery.orgchart.js]: I can't add this comment to your screenshot/image - but it does not look like the other ones.

File: demo/simple.html (hunk 2 above)
Comment (Goback) [inline demo/simple.html]: yes,it's true,
read me updated about demo
link to live demo can be on your site :)

File: demo/simple.html (hunk 3 above)
Comment (Goback) [inline demo/simple.html]: you are true
nothing really.

File: jquery.orgchart.css (hunk 4 above)
Comment (Goback) [inline jquery.orgchart.css]: I'll correct it

File: jquery.orgchart.css (hunk 5 above)
Comment (Goback) [inline jquery.orgchart.css]: it's break rtl flow,so was added in another line(just for ltr)
```
div.orgChart.ltr div.adjunct.node{	margin-left		: -110px; }
```

File: jquery.orgchart.js (hunk 12 above)
Comment (Goback) [inline jquery.orgchart.js]: "
Don't remove the newline at the EOF.
"
Why?

**Comment by caprica** [comment]
Closing since all future work will go into the react-org-chart project instead.
